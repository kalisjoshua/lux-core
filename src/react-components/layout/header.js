/** @memberof module:react-components/Layout */

import React from 'react'; // Even if this isn't used it needs to be imported!

function headerComponent() {

  return (
    <header className="main-header" role="banner">
      {/* <!-- logo link --> */}

      {/* <!-- main navigation --> */}

      {/* <!-- auth link --> */}
    </header>
  );
}
headerComponent.propTypes = {
};

export default headerComponent;
