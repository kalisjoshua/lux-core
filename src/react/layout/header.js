/**
 * @module Header
 * @memberof lux-react/layout
 */

import React from 'react'; // `React` must be in scope when using JSX

import { hasOne } from '../../lib/has';

import './auth';
import './navigation';

import registry from '../componentRegistry';
import shapeOfSiren from '../siren.react';

/**
 * Creates the main header of the site layout.
 *
 * @param  {ReactComponentProps} props - A Siren object from the server.
 * @param  {array} props.links - A collection of link objects related to the
 * current resource.
 *
 * @return {ReactComponent}
 *
 * ## Siren Considerations
 *
 * The `Header` component makes use of the Siren properties: `links`, and
 * `actions`. The `links` are searched for the API "root" URI (resource)
 * indicated by the `rel` "index". The `links` are also passed to the
 * `Navigation` component for generating the main menu of the page. The Siren
 * property `actions` is searched for an action for "loggin in/out"; if neither
 * are found the auth-link is omitted completely.
 *
 * ### Siren Property & Usage
 *
 *   - `links`
 *     1. Header banner link for application
 *     2. Main menu, navigation, links
 *   - `actions`
 *     1. Login/logout - action `name`
 */
function headerComponent(props) {
  const rootLink = (props.links || [])
    .filter(l => hasOne('rel', 'index', l))[0] || {};

  const fallbackLink = {
    href: rootLink.href || '/',
    rel: ['section'],
    title: rootLink.title || 'Home',
  };

  const { href, title } = fallbackLink;

  const Auth = registry('Auth');
  const Navigation = registry('Navigation');

  return (
    <header className="main-header" role="banner">
      <h1><a className="link--heroOrSomething" href={href}>{title}</a></h1>

      <Navigation links={props.links || [fallbackLink]} />

      <Auth links={props.links || []} />
    </header>
  );
}
headerComponent.propTypes = {
  ...shapeOfSiren
};

registry('Header', headerComponent);
export default registry('Header');
