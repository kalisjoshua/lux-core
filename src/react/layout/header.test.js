import React from 'react';
import renderer from 'react-test-renderer';

import registry from '../componentRegistry';

import './header';
const Header = registry('Header');

describe('Header (supplied Layout)', function () {
  it('should exist; and should be a function', function () {
    expect(typeof Header).toMatch(/function/i);
  });

  it('should match the (default) snapshot', function () {
    const component = renderer.create(
      <Header />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should pull the link with "index" for a rel as main link', function () {
    const links = [
      { href: '/', rel: ['index'], title: 'root' },
    ];
    const component = renderer.create(
      <Header links={links}></Header>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
