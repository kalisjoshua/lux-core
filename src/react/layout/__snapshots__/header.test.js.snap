exports[`Header (supplied Layout) should match the (default) snapshot 1`] = `
<header
  className="main-header"
  role="banner">
  <h1>
    <a
      className="link--heroOrSomething"
      href="/">
      Home
    </a>
  </h1>
  <navigation
    id="mainnavigation"
    role="navigation">
    <ul />
  </navigation>
  <noscript />
</header>
`;

exports[`Header (supplied Layout) should pull the link with "index" for a rel as main link 1`] = `
<header
  className="main-header"
  role="banner">
  <h1>
    <a
      className="link--heroOrSomething"
      href="/">
      root
    </a>
  </h1>
  <navigation
    id="mainnavigation"
    role="navigation">
    <ul />
  </navigation>
  <noscript />
</header>
`;
